yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -372,11 +381,29 @@ private[spark] class Client(
       sparkConf.set(CONF_SPARK_YARN_SECONDARY_JARS, cachedSecondaryJarLinks.mkString(","))
     }
 
+    if (args.primaryPyFile != null) {
+      distribute(args.primaryPyFile)
```

Comment:
Hmm, strangely enough it seems things work without distributing the main file. Wonder if that's safe in all cases or just a side-effect of the really simple test code, though.